File: android移动开发/期末课设/Company/app/src/main/java/com/example/company/NewsActivity.java
package com.example.company;

import androidx.appcompat.app.AppCompatActivity;

import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.os.Bundle;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.Objects;

public class NewsActivity extends AppCompatActivity {

    public static final String EXTRA_DRINKID = "newsId";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news);

        int newsId = Objects.requireNonNull(getIntent().getExtras()).getInt(EXTRA_DRINKID);

        SQLiteOpenHelper starbuzzDatabaseHelper = new CompanyDatabaseHelper(this);
        //获得数据库引用
        try (SQLiteDatabase db = starbuzzDatabaseHelper.getReadableDatabase()) {
            Cursor cursor = db.query("NEWS",
                    new String[]{"TITLE", "IMAGE_RESOURCE_ID", "CONTENT"},
                    "_id=?",
                    new String[]{Integer.toString(newsId)},
                    null,
                    null,
                    null);
            if (cursor.moveToFirst()) {

                String titleText = cursor.getString(0);
                int photoId = cursor.getInt(1);
                String contentText = cursor.getString(2);
                TextView name = findViewById(R.id.title);
                name.setText(titleText);

                ImageView imagenews = findViewById(R.id.image_news);
                imagenews.setImageResource(photoId);
                imagenews.setContentDescription(titleText);

                TextView content = findViewById(R.id.content);
                content.setText(contentText);
            }
            cursor.close();
        } catch (SQLException e) {
            Log.e("sqlite", Objects.requireNonNull(e.getMessage()));
            Toast toast = Toast.makeText(this, "Database unavailable", Toast.LENGTH_SHORT);
            toast.show();
        }
    }

}

File: android-mobile-development/final_course_project/Company/app/src/main/java/com/example/company/ui/news/NewsActivity.kt
package com.example.company.ui.news

import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.example.company.R
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class NewsActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_NEWS_ID = "newsId"
    }

    private val viewModel: NewsViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news)

        val newsId = intent.getIntExtra(EXTRA_NEWS_ID, -1)
        if (newsId == -1) {
            Toast.makeText(this, "新闻不存在", Toast.LENGTH_SHORT).show()
            finish()
            return
        }

        viewModel.loadNews(newsId)

        viewModel.news.observe(this) { news ->
            if (news != null) {
                findViewById<TextView>(R.id.title).text = news.title
                findViewById<ImageView>(R.id.image_news).apply {
                    setImageResource(news.imageResourceId)
                    contentDescription = news.title
                }
                findViewById<TextView>(R.id.content).text = news.content
            }
        }
    }
}
